### DISCUSSION TITLE
TypeError: sdepth() missing 1 required positional argument: 'C'

### DISCUSSION THREAD

**Discussion by brucelaughlin** [OpenDrift/opendrift#1395] [General]
Haven't seen this before.  The run is proceeding, but I'm not sure whether I should be concerned.  Please advise!

 21 00:26:58 INFO    opendrift.models.basemodel:947: All points are in ocean
 22 00:26:58 WARNING opendrift.models.basemodel:730: Seafloor check not being run because environment is missing. This will happen the first time the function is run but if it happens subsequently there is probably a problem.
 23 00:26:58 INFO    opendrift.models.basemodel:2038: 1990-01-01 12:00:00 - step 1 of 4608 - 11844 active elements (0 deactivated)
 24 00:26:58 INFO    opendrift.models.basemodel.environment:665: ========================
 25 00:26:58 ERROR   opendrift.models.basemodel.environment:666: sdepth() missing 1 required positional argument: 'C'
 26 Traceback (most recent call last):
 27   File "/home/blaughli/opendrift/opendrift/models/basemodel/environment.py", line 641, in get_environment
 28     reader.get_variables_interpolated(
 29   File "/home/blaughli/opendrift/opendrift/readers/basereader/variables.py", line 904, in get_variables_interpolated
 30     env, env_profiles = self.get_variables_interpolated_xy(
 31                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 32   File "/home/blaughli/opendrift/opendrift/readers/basereader/variables.py", line 770, in get_variables_interpolated_xy
 33     env, env_profiles = self._get_variables_interpolated_(
 34                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 35   File "/home/blaughli/opendrift/opendrift/readers/basereader/structured.py", line 281, in _get_variables_interpolated_
 36     self.get_variables(blockvariables_before, time_before,
 37   File "/home/blaughli/tracking_project/opendrift_custom/readers/reader_ROMS_native_custom_eco.py", line 376, in get_variables
 38     self.z_rho_tot = depth.sdepth(Htot, self.hc, self.Cs_r,
 39                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 40 TypeError: sdepth() missing 1 required positional argument: 'C'
 41 00:26:58 INFO    opendrift.models.basemodel.environment:668: ========================
 42 00:26:58 INFO    opendrift.models.basemodel:2038: 1990-01-01 13:00:00 - step 2 of 4608 - 11844 active elements (0 deactivated)
 43 00:26:58 INFO    opendrift.models.basemodel.environment:665: ========================
 44 00:26:58 ERROR   opendrift.models.basemodel.environment:666: sdepth() missing 1 required positional argument: 'C'
 45 Traceback (most recent call last):
 46   File "/home/blaughli/opendrift/opendrift/models/basemodel/environment.py", line 641, in get_environment
 47     reader.get_variables_interpolated(
 48   File "/home/blaughli/opendrift/opendrift/readers/basereader/variables.py", line 904, in get_variables_interpolated
 49     env, env_profiles = self.get_variables_interpolated_xy(
 50                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 51   File "/home/blaughli/opendrift/opendrift/readers/basereader/variables.py", line 770, in get_variables_interpolated_xy
 52     env, env_profiles = self._get_variables_interpolated_(
 53                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 54   File "/home/blaughli/opendrift/opendrift/readers/basereader/structured.py", line 281, in _get_variables_interpolated_
 55     self.get_variables(blockvariables_before, time_before,
 56   File "/home/blaughli/tracking_project/opendrift_custom/readers/reader_ROMS_native_custom_eco.py", line 376, in get_variables
 57     self.z_rho_tot = depth.sdepth(Htot, self.hc, self.Cs_r,
 58                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 59 TypeError: sdepth() missing 1 required positional argument: 'C'
 60 00:26:58 INFO    opendrift.models.basemodel.environment:668: ========================
 61 00:26:58 WARNING opendrift.models.basemodel.environment:673: Reader roms native is discarded after failing more times than allowed (1)
 62 00:26:58 INFO    opendrift.models.basemodel:2038: 1990-01-01 14:00:00 - step 3 of 4608 - 11844 active elements (0 deactivated)

**Comment by gauteh** [comment]
Yes, this is a bit worrying. OpenDrift tries to keep going, but this appears to be a bug in the ROMS reader.

**Comment by knutfrode** [comment]
Could you check (`git status`) if your repository have been corrupted somehow?
Which version are you using?
If everything is ok with code, could you share such a sample ROMS file?

**Comment by brucelaughlin** [comment]
Apologies, I was using code based on an older version of the reader.  This seems to have been resolved now!  Sorry to take your time, thanks for your attention.
